src/xdist/remote.py

```diff
@@ -201,30 +201,25 @@ def run_one_test(self) -> None:
             "runtest_protocol_complete", item_index=self.item_index, duration=duration
         )
 
-    def pytest_collection_modifyitems(
-        self,
-        config: pytest.Config,
-        items: list[pytest.Item],
-    ) -> None:
-        # add the group name to nodeid as suffix if --dist=loadgroup
-        if config.getvalue("loadgroup"):
-            for item in items:
-                mark = item.get_closest_marker("xdist_group")
-                if not mark:
-                    continue
-                gname = (
-                    mark.args[0]
-                    if len(mark.args) > 0
-                    else mark.kwargs.get("name", "default")
-                )
-                item._nodeid = f"{item.nodeid}@{gname}"
-
     @pytest.hookimpl
     def pytest_collection_finish(self, session: pytest.Session) -> None:
+        # collect the scope for each node for --dist=loadgroup
+        loadgroup_scopes = {}
+        for item in session.items:
+            mark = item.get_closest_marker("xdist_group")
+            if not mark:
+                continue
+            gname = (
+                mark.args[0]
+                if len(mark.args) > 0
+                else mark.kwargs.get("name", "default")
+            )
+            loadgroup_scopes[item.nodeid] = gname
```

Comment:
An important Note here is that unfortunately duplicate node ids are only discouraged,not enforced 

Which is the reason for using the node index instead of the node id when communicating using the scheduler

Reply:
That's pretty bad. What's your opinion, does this need to be addressed, or can we live with duplicate node ids being scheduled on the same node, per `xdist_group`, even only one test had the annotation?